src/main/java/binnie/genetics/machine/isolator/IsolatorComponentLogic.java

```diff
@@ -83,22 +78,14 @@ protected void onFinishTask() {
         }
 
         final IChromosomeType[] karyo = root.getKaryotype();
-        final IChromosomeType chromosome = karyo[rand.nextInt(karyo.length)];
-        final IAllele allele;
-        try {
-            allele = rand.nextBoolean() ? individual.getGenome().getActiveAllele(chromosome)
-                    : individual.getGenome().getInactiveAllele(chromosome);
-        } catch (NullPointerException npe) {
-            // just return and eat the power without eating the rest of it
-            LOG.warn(
-                    "NPE occurred on unknown gene: name: " + chromosome.getName()
-                            + " | ordinal: "
-                            + chromosome.ordinal()
-                            + " | is null? "
-                            + (individual.getGenome().getChromosomes()[chromosome.ordinal()] == null ? "true"
-                                    : "false"));
-            return;
-        }
+
+        IChromosomeType chromosome;
+        do {
+            chromosome = karyo[rand.nextInt(karyo.length)];
+        } while (individual.getGenome().getChromosomes()[chromosome.ordinal()] == null);
```

Comment:
Is it possible for all chromosomes to be null here, causing an infinite loop?

Reply:
It would be impossible for normal playthrough. 
If all chromosomes are null it means that the be doesn't even have a species on them. So, i wouldn't see anyone puting a blank bee in the isolator nor obtaining them.
Nonetheless, I think it would be nice to have a some kind of way to prevent an infinite loop. Incase someone puts a blank bee for debbuging purpose.

Reply:
How about something like this?
```java
List<IChromosomeType> chromosomes = Arrays.stream(karyo)
    .filter(Objects::nonNull)
    .collect(Collectors.toList());
if (chromosomes.isEmpty()) return;
IChromosomeType chromosome = chromosomes.get(rand.nextInt(chromosomes.size()));
```

Reply:
karyo is a list that contains the chromosome types (SPECIES, PRODUCTION, HUMIDITY etc)

So, if we change this part of line 1 
Arrays.stream(karyo) 
to 
Arrays.stream(individual.getGenome().getChromosomes()) 

And maybe change the last line too accordingly

So if we fix that part it would be a much better solution : D


* I'm on my phone so excuse me for poor formatting.